doc/CompilationPython.md

```diff
@@ -0,0 +1,189 @@
+| [Tutorials Home](index.md) | [Previous](ImportExport.md) | [Next](PythonModule.md) |
+| :--- | :---: | ---: |
+
+# Compiling libpointmatcher with python
+
+This tutorial presents the different steps of compiling pypointmatcher, the libpointmatcher's python module, on Ubuntu and Mac OS X.
+
+## Prerequisites
+
+To get started, you will obviously need the same prerequisites as libpointmatcher, but also some additional dependencies as listed here :
+
+| Name | Version <br> (Tested August 2020 on Ubuntu 18.04) |
+| :--- | :---: |
+| pybind11 | 2.5.0 |
+| Python3 | 3.6.9 |
+|||
+| **Dependencies** | |
+| catch | 1.10.0 |
+| python3-dev | 3.6.7 |
+| pytest | 5.4.3 |
+
+`catch` and `python3-dev` need to be installed with the package manager :
+
+```bash
+sudo apt install catch python3-dev
+```
+
+and `pytest` with `pip` :
+
+```bash
+pip3 install pytest
+```
+
+> ***Note :*** pybind11 is also compatible with Python2.7, which means that it is also possible to generate a module for Python2.7. But, this hasn't been tested. You can refer to the pybind11 [official documentation](https://pybind11.readthedocs.io/en/latest/index.html) for more informations on what you can do with pybind11 and how to do it.
+
+The rest of this tutorial will guide you through the necessary steps to compile pypointmatcher.
+
+## pybind11
+
+In order to be able to compile pypointmatcher, you must either install pybind11 on your system or add it as a git submodule in the libpointmatcher's `/contrib` directory. You must then create a symbolic link to this git submodule in the python directory. Go [here](#installing-pybind11) for the installation steps or [here](#adding-pybind11) for the git sudmodule steps.
+
+### Installing pybind11 (recommended) <a name="installing-pybind11"></a>
+
+The very first step is to clone pybind11 [repository](https://github.com/pybind/pybind11) into a directory of your choice.
+ 
+ At the moment, pypointmatcher can only be compiled with **version 2.5.0** of pybind11. To install the right version, run the following commands :
+ 
+ ```bash
+ cd pybind11
+ git checkout v2.5.0
+ ```
+ 
+Once this is done, run the following commands :
+
+```bash
+mkdir build && cd build
+cmake ..
+make check -j 4
+```
+
+This will both compile and run the pybind11 tests. Next, you can install pybind11 by running this command :
+
+```bash
+sudo make install
```

Comment:
Where does this install the bindings? Usual /usr/lib, or /home/<user>/.local/python3x/ ?

The second is the usual one for packages fetched via pip.

Reply:
This will install pybind11, not the bindings, in the `/usr/local/include` directory. It is up to the user to specify where the bindings will be installed. If no location are specified, the default path will be `/usr/local/lib/python3.x/dist-packages` for the system python environment, which is not recommended. Instead, it should be `/home/username/.local/lib/python3.x/site-packages` , like you said. And it would something like `/home/username/python_virtual_env_dir/virtual_env_name/lib/python3.x/site-packages` for a virtual python environment. I think this is well explained in the tutorial. I could add another warning about not using the `dist-packages` directory, since it is reserved for packages installed with the system package manager. At least for Linux system AFAIK.

Reply:
I've checked where libnabo python bindings were installed in order to take a decision. I have the python2 and python3 versions installed on my computer. The python2 version is installed in the `/usr/local/lib/python2.7/dist-packages` directory and the python3 version is installed in the `/usr/local/lib/python3.6/dist-packages` and in the `/home/username/.local/lib/python3.6/site-packages` directory. So, what would be the best option?